File: SDKs/microsoft.com/cognitiveservices-Training/3.1/typescript/src/sdk/models/shared/iteration.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum IterationClassificationTypeEnum {
    Multiclass = "Multiclass",
    Multilabel = "Multilabel"
}

export enum IterationExportableToEnum {
    CoreMl = "CoreML",
    TensorFlow = "TensorFlow",
    DockerFile = "DockerFile",
    Onnx = "ONNX",
    Vaidk = "VAIDK"
}

export enum IterationTrainingTypeEnum {
    Regular = "Regular",
    Advanced = "Advanced"
}


// Iteration
/** 
 * Iteration model to be sent over JSON.
**/
export class Iteration extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=classificationType" })
  classificationType?: IterationClassificationTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=created" })
  created?: Date;

  @SpeakeasyMetadata({ data: "json, name=domainId" })
  domainId?: string;

  @SpeakeasyMetadata({ data: "json, name=exportable" })
  exportable?: boolean;

  @SpeakeasyMetadata({ data: "json, name=exportableTo" })
  exportableTo?: IterationExportableToEnum[];

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=lastModified" })
  lastModified?: Date;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "json, name=originalPublishResourceId" })
  originalPublishResourceId?: string;

  @SpeakeasyMetadata({ data: "json, name=projectId" })
  projectId?: string;

  @SpeakeasyMetadata({ data: "json, name=publishName" })
  publishName?: string;

  @SpeakeasyMetadata({ data: "json, name=reservedBudgetInHours" })
  reservedBudgetInHours?: number;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: string;

  @SpeakeasyMetadata({ data: "json, name=trainedAt" })
  trainedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=trainingTimeInMinutes" })
  trainingTimeInMinutes?: number;

  @SpeakeasyMetadata({ data: "json, name=trainingType" })
  trainingType?: IterationTrainingTypeEnum;
}


// IterationInput
/** 
 * Iteration model to be sent over JSON.
**/
export class IterationInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=name, form, name=name;" })
  name: string;
}

File: SDKs/microsoft.com/cognitiveservices-Training/3.2/python/src/sdk/models/shared/iteration.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class IterationClassificationTypeEnum(str, Enum):
    MULTICLASS = "Multiclass"
    MULTILABEL = "Multilabel"

class IterationExportableToEnum(str, Enum):
    CORE_ML = "CoreML"
    TENSOR_FLOW = "TensorFlow"
    DOCKER_FILE = "DockerFile"
    ONNX = "ONNX"
    VAIDK = "VAIDK"

class IterationTrainingTypeEnum(str, Enum):
    REGULAR = "Regular"
    ADVANCED = "Advanced"


@dataclass_json
@dataclasses.dataclass
class Iteration:
    r"""Iteration
    Iteration model to be sent over JSON.
    """
    
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    classification_type: Optional[IterationClassificationTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('classificationType') }})
    created: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    domain_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('domainId') }})
    exportable: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('exportable') }})
    exportable_to: Optional[list[IterationExportableToEnum]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('exportableTo') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    last_modified: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lastModified'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    original_publish_resource_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('originalPublishResourceId') }})
    project_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('projectId') }})
    publish_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('publishName') }})
    reserved_budget_in_hours: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('reservedBudgetInHours') }})
    status: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    trained_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('trainedAt'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    training_time_in_minutes: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('trainingTimeInMinutes') }})
    training_type: Optional[IterationTrainingTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('trainingType') }})
    

@dataclass_json
@dataclasses.dataclass
class IterationInput:
    r"""IterationInput
    Iteration model to be sent over JSON.
    """
    
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }, 'form': { 'field_name': 'name' }})
    
